core/input/input_event.cpp

```diff
@@ -1356,7 +1356,18 @@ bool InputEventAction::action_match(const Ref<InputEvent> &p_event, bool p_exact
 }
 
 String InputEventAction::as_text() const {
-	return vformat(RTR("Input Action %s was %s"), action, pressed ? "pressed" : "released");
+	const List<Ref<InputEvent>> *events = InputMap::get_singleton()->action_get_events(action);
+	if (!events) {
+		return String();
+	}
+
+	for (const List<Ref<InputEvent>>::Element *E = events->front(); E; E = E->next()) {
+		if (E->get().is_valid()) {
+			return E->get()->as_text();
+		}
+	}
```

Comment:
Couldn't this loop be simplified to:
``` C++
	for (Ref<InputEvent> E : *events) {
		if (E.is_valid()) {
			return E->as_text();
		}
	}
```

Reply:
Yep, used the old way for some reason here...